tools/filesystem/file_test.go

```diff
@@ -43,6 +43,33 @@ func TestNewFileFromFromPath(t *testing.T) {
 	}
 }
 
+func TestNewFileFromBytes(t *testing.T) {
+	testDir := createTestDir(t)
+	defer os.RemoveAll(testDir)
+
+	// nil bytes
+	if _, err := filesystem.NewFileFromBytes(nil, "photo.jpg"); err == nil {
+		t.Fatal("Expected error, got nil")
+	}
+
+	// zero bytes
+	if _, err := filesystem.NewFileFromBytes([]byte{}, "photo.jpg"); err == nil {
+		t.Fatal("Expected error, got nil")
+	}
+
+	originalName := "notes.txt"
+	f, err := filesystem.NewFileFromBytes([]byte("TODO: Write Tests\n"), originalName)
```

Comment:
Can we change the text to just `test` or something else shorter?
_(Additionally, I also have a regex pattern for plain-tasks that extracts "todo" strings from the code and will get triggered unnecessary)_